carta/image.py

```diff
@@ -605,6 +605,189 @@ def hide_contours(self):
         """Hide the contours."""
         self.set_contours_visible(False)
 
+    # VECTOR OVERLAY
+
+    @validate(NoneOr(Constant(VectorOverlaySource)), NoneOr(Constant(VectorOverlaySource)), NoneOr(Number()), Boolean(), NoneOr(Number()), NoneOr(Number()), NoneOr(Number()))
+    def configure_vector_overlay(self, angular_source=None, intensity_source=None, pixel_averaging=4, fractional_intensity=False, threshold=None, q_error=None, u_error=None):
+        """Configure vector overlay.
+
+        Parameters
+        ----------
+        angular_source : {0}
+            The angular source. If the image is with Stoke information, ``Computed PA`` is set by default; If the image is without Stoke information, ``Current image`` is set.
+        intensity_source : {1}
+            The intensity source. If the image is with Stoke information, ``Computed PI`` is set by default; If the image is without Stoke information, ``Current image`` is set.
+        pixel_averaging : {2}
+            The pixel averaging width in pixel. Set to ``None`` to disable pixel averaging.
+        fractional_intensity : {3}
+            Enable fractional polarization intensity. By default the absolute polarization intensity is used.
+        threshold : {4}
+            The threshold in Jy/pixels. By default is 4.
+        q_error : {5}
+            The Stokes Q error in Jy/beam. Set both this and ``u_error`` to enable debiasing. Debiasing is disabled by default.
+        u_error : {6}
+            The Stokes U error in Jy/beam. Set both this and ``q_error`` to enable debiasing. Debiasing is disabled by default.
+        """
+        pixel_averaging_enabled = pixel_averaging is not None
+        threshold_enabled = threshold is not None
+        debiasing = q_error is not None and u_error is not None
+        if not debiasing and (q_error is not None or u_error is not None):
+            logger.warning("The Stokes Q and the Stokes U must both be set to enable debiasing.")
+        if angular_source is None:
+            angular_source = "None"
+        if intensity_source is None:
+            intensity_source = "None"
+        if threshold is None:
+            threshold = "None"
+        if q_error is None:
+            q_error = "None"
+        if u_error is None:
+            u_error = "None"
```

Comment:
All of these `"None"`s need to be replaced by macros -- this hack should *only* be used when it's necessary to explicitly *unset* a property (which is only the case with the intensity range values). In this case we *don't* want to unset these properties if the parameters are `None`; we want to keep the properties which are currently set in the frontend.